components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/dao/impl/APIResourceManagementDAOImpl.java

```diff
@@ -1087,6 +1087,54 @@ private void addScopes(Connection dbConnection, String apiId, List<Scope> scopes
         }
     }
 
+    /**
+     * Add scopes to the API resource.
+     *
+     * @param dbConnection Database connection.
+     * @param apiId        API resource id.
+     * @param scopes       List of scopes.
+     * @param tenantId     Tenant id.
+     * @throws APIResourceMgtException If an error occurs while adding scopes.
+     */
+    private void addScopes(Connection dbConnection, String apiId, List<Scope> scopes, Integer tenantId,
```

Comment:
We can remove the previous method `addScopes`.
From the calling methods we can pass the resource type. If not able to find we can pass null so that logic will be executed as previously